include/revng/FunctionIsolation/EnforceABI.h

```diff
@@ -10,14 +10,25 @@
 
 #include "revng/BasicAnalyses/GeneratedCodeBasicInfo.h"
 
-class EnforceABI : public llvm::ModulePass {
+class EnforceABIPassImpl : public pipeline::FunctionPassImpl {
+
 public:
-  static char ID;
+  using pipeline::FunctionPassImpl::FunctionPassImpl;
+  bool prologue(llvm::Module &M, const model::Binary &Model) override;
+  bool runOnFunction(llvm::Function &Function,
+                     const model::Function &ModelFunction) override {
+    return false;
+  }
+
+private:
+};
 
+class EnforceABI : public pipeline::FunctionPass<EnforceABIPassImpl> {
 public:
-  EnforceABI() : ModulePass(ID) {}
+  static char ID;
 
-  bool runOnModule(llvm::Module &M) override;
+public:
+  EnforceABI() : pipeline::FunctionPass<EnforceABIPassImpl>(ID) {}
 
   void getAnalysisUsage(llvm::AnalysisUsage &AU) const override;
 };
```

Comment:
This has to go. It does nothing except inheriting from `pipeline::FunctionPass<EnforceABIPassImpl>`.
We already have to define the pipe and the `FunctionPassImpl`, not the `FunctionPass` too.
We can push `getAnalysisUsage` into `FunctionPassImpl` too, if that's the reason.